packages/backend/server/src/__tests__/models/workspace-page.spec.ts

```diff
@@ -0,0 +1,222 @@
+import { TestingModule } from '@nestjs/testing';
+import { PrismaClient } from '@prisma/client';
+import ava, { TestFn } from 'ava';
+
+import { Config } from '../../base/config';
+import { Permission, PublicPageMode } from '../../core/permission/types';
+import { UserModel } from '../../models/user';
+import { WorkspaceModel } from '../../models/workspace';
+import { WorkspacePageModel } from '../../models/workspace-page';
+import { createTestingModule, initTestingDB } from '../utils';
+
+interface Context {
+  config: Config;
+  module: TestingModule;
+  db: PrismaClient;
+  user: UserModel;
+  workspace: WorkspaceModel;
+  workspacePage: WorkspacePageModel;
+}
+
+const test = ava as TestFn<Context>;
+
+test.before(async t => {
+  const module = await createTestingModule({
+    providers: [UserModel, WorkspaceModel],
+  });
+
+  t.context.user = module.get(UserModel);
+  t.context.workspace = module.get(WorkspaceModel);
+  t.context.workspacePage = module.get(WorkspacePageModel);
+  t.context.db = module.get(PrismaClient);
+  t.context.config = module.get(Config);
+  t.context.module = module;
+});
+
+test.beforeEach(async t => {
+  await initTestingDB(t.context.db);
+});
+
+test.after(async t => {
+  await t.context.module.close();
+});
+
+test('should create workspace page with default mode and public false', async t => {
+  const user = await t.context.user.create({
+    email: 'test@affine.pro',
+  });
+  const workspace = await t.context.workspace.create(user.id);
+  const page = await t.context.workspacePage.createOrUpdate(
```

Comment:
im wondering is it good to call it `page` instead. save my keyboard [![bow](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/WjlTemxb6oA4PgZFmj08/d71dd07a-f2ef-43c2-a156-72417d63807a/bow.gif)](https://app.graphite.dev//settings/meme-library?org=toeverything)

Reply:
Sure, I will change `WorkspacePageModel` to `PageModel`.

[![approveyoda](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/WjlTemxb6oA4PgZFmj08/961a214d-0cdd-4a5e-a5c1-6cef8cfb0b05/approveyoda.jpeg)](https://app.graphite.dev//settings/meme-library?org=toeverything)